### ISSUE TITLE
As an admin with access to all four pilot forests I want to be able to run a report on permit sales for multiple forests at the same time.

### ISSUE THREAD

**Issue by ASprinkle** [18F/fs-open-forest-platform#484]
## Notes
Currently if we want to know total permit sales in a certain time frame, we have to run separate reports and add the totals manually.

## Acceptance Criteria
- [x] An Open Forest administrator with roles for all four forests can run a single report that shows total permits issued, trees, and value for each forest.
- [x] A report for multiple forests should show differentiation between forests (i.e., data would be separated by Forest Name).

## Tasks
- [x] Add new server route to query all for all forests
- [x] Ensure admin authentication is for all four forests
- [x] Update the frontend to include a column or row of which forest the data is generated for all forests (and potentially design)
- [x] Ensure the .csv download also has the forest names indicated
-  Format the generated report so that multiple forests' data are respectively divided.
- [x] PO approved.

## Definition of Done
- [x] Code unit / e2e tested and comply with QWASP
- [x] Server actions captured by logs
- [x] No new security vulnerabilities
- [x] Internal team code review
- [x] Documentation / readme.md / CircleCI updated
- [x] Docker updated
- [x] Compare finished design with mockup
- [x] Usability tested

**Comment by ASprinkle** [comment]
Estimate at refinement

**Comment by ASprinkle** [comment]
@davemcorwin is it reasonable to add an AC to this that says : "A report for multiple forests would show differentiation between forests (i.e., data would be separated by Forest Name)."?

**Comment by ASprinkle** [comment]
From stand up: @davemcorwin will post some questions regarding time zones for pulling multiple forest reports.

**Comment by davemcorwin** [comment]
@ASprinkle Wanted to discuss how we want to handle timezones when querying across multiple forests. The current implementation only queries for 1 forest at a time and very specifically uses the timezone of the selected forest when:
1. determining the start end dates for the UI
2. displaying start and end dates in the UI
3. querying for permits on the backend

To support this feature, we query across multiple forests which may be in different timezones so we must decide how to handle this. I know there were some previous discussions regarding timezones so want to loop in @adborden in case he has some context that would be helpful.

My suggestion is the following:
- When determining and displaying the dates in the UI use the minimum and maximum dates from the authorized forests, ignoring timezones.
- When querying for the permits in the database we have a couple options:
  1, Go a bit deeper into SQL and query using each forests' timezone or
  2. choose a single timezone to query against either accepting the possibility that some permits will be included in an incorrect reporting period according the forest timezone.

If its IMPORTANT for the reports to exactly reflect the permits purchased between the requested dates according to each forests' timezone, I recommend we do #1 above.

3. compute a minimum and maximum date in a particular timezone to query against permits for all authorized forests

My thought is that the most important thing is to be consistent and repeatable, so even if a permit is  misreported according to its forst timezone, it is not double-counted or ignored.

**Comment by ASprinkle** [comment]
@davemcorwin It is important for reports to exactly reflect the permits purchased between the requested dates according to each forests' timezone, because each forest may have a specific reporting period or reporting expectation.

Note: This timezone question is also linked to story #468 

I agree that consistency is important, and that we not double count or ignore data.

**Comment by ASprinkle** [comment]
From stand up: waiting on #483 and #468 to complete first

**Comment by ASprinkle** [comment]
![image](https://user-images.githubusercontent.com/33435621/53504409-c2283180-3a66-11e9-82dc-8073315a5605.png)

It looks like it works really well in staging!  Having the option to download as excel is really helpful for this so that the admin can sort and filter if the wanted to.

**Comment by davemcorwin** [comment]
If possible, I would like to save the formatting for another issue. We can make the column larger to accomodate longer names or truncate the name (Ex. `Arapaho-Ro...`)

**Comment by ASprinkle** [comment]
Ok that sounds good. thanks @davemcorwin
